---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# daeqtlr

<!-- badges: start -->

<!-- badges: end -->

The goal of `{daeqtlr}` is to perform differential allelic expression trait loci
(DAEQTL) mapping.

Differential allelic expression (DAE) can be treated as a quantitative trait.
Marker genotypes collected from the same set of individuals can, in turn, be
used in testing statistical associations with DAE. This package provides
routines for performing such a workflow, i.e. DAEQTL mapping.

Learn more in <https://maialab.org/daeqtlr>.

## Installation

You can install the development version of `{daeqtlr}` like so:

``` r
# install.packages("remotes")
remotes::install_github("maialab/daeqtlr")
```

## Usage

Here's a quick example on how to perform DAEQTL mapping with `{daeqtlr}`. For
more details about this example read `vignette("daeqtlr")`.

```{r}
library(daeqtlr)

# SNP pairs to be tested for statistical association
snp_pairs <- read_snp_pairs(file = daeqtlr_example("snp_pairs.csv"))

# Zygosity levels (homozygous or heterozygous) for genotyped SNPs
zygosity <- read_snp_zygosity(file = daeqtlr_example("zygosity.csv"))

# Allelic expression (AE) ratios of DAE SNPs
ae <- read_ae_ratios(file = daeqtlr_example("ae.csv"))

mapping_dt <-
  daeqtl_mapping(snp_pairs = snp_pairs,
                 zygosity = zygosity,
                 ae = ae)

# Here's the first 30 SNP pairs tested
mapping_dt[1:30, -c('dae_snp_position', 'candidate_snp_position')]
```
