<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function implements daeqtlr's default approach for testing the
association between a candidate SNP zygosity and the allelic expression of a
DAE SNP. The method takes into account the pattern of the allelic expression
(AE) ratios' distribution displayed at each DAE SNP (depicted below), as this
is dependent on the linkage disequilibrium between the DAE SNP and the
candidate SNP. In some cases the requirements for application of this
methodology are not met and no measure of statistical association
significance is derived. These cases are depicted in the figure below, and
further explained in the section Value.
"><title>DAEQTL test — daeqtl_test • daeqtlr</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="DAEQTL test — daeqtl_test"><meta property="og:description" content="This function implements daeqtlr's default approach for testing the
association between a candidate SNP zygosity and the allelic expression of a
DAE SNP. The method takes into account the pattern of the allelic expression
(AE) ratios' distribution displayed at each DAE SNP (depicted below), as this
is dependent on the linkage disequilibrium between the DAE SNP and the
candidate SNP. In some cases the requirements for application of this
methodology are not met and no measure of statistical association
significance is derived. These cases are depicted in the figure below, and
further explained in the section Value.
"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">daeqtlr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/daeqtlr.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/genotypes-to-zygosity.html">From genotypes to zygosity</a>
    <a class="dropdown-item" href="../articles/snp-pairs.html">Preparing the set of SNP pairs for DAEQTL mapping</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/maialab/daeqtlr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>DAEQTL test</h1>
      <small class="dont-index">Source: <a href="https://github.com/maialab/daeqtlr/blob/HEAD/R/daeqtl_test.R" class="external-link"><code>R/daeqtl_test.R</code></a></small>
      <div class="d-none name"><code>daeqtl_test.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function implements daeqtlr's default approach for testing the
association between a candidate SNP zygosity and the allelic expression of a
DAE SNP. The method takes into account the pattern of the allelic expression
(AE) ratios' distribution displayed at each DAE SNP (depicted below), as this
is dependent on the linkage disequilibrium between the DAE SNP and the
candidate SNP. In some cases the requirements for application of this
methodology are not met and no measure of statistical association
significance is derived. These cases are depicted in the figure below, and
further explained in the section Value.</p>
<p><img src="figures/mapping_approach.svg" alt="Mapping approach"></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">daeqtl_test</span><span class="op">(</span>
  <span class="va">ae_hom</span>,
  <span class="va">ae_het</span>,
  dae_threshold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></span><span class="op">(</span><span class="fl">1.5</span><span class="op">)</span>,
  min_n_hom <span class="op">=</span> <span class="fl">2L</span>,
  min_n_het <span class="op">=</span> <span class="fl">2L</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>ae_hom</dt>
<dd><p>Numeric vector of AE ratios of the DAE SNP. Each element of the
vector refers to a sample that is homozygous (<code>hom</code>) for the candidate
SNP.</p></dd>
<dt>ae_het</dt>
<dd><p>Numeric vector of AE ratios of the DAE SNP. Each element of the
vector refers to a sample that is heterozygous (<code>het</code>) for the
candidate SNP.</p></dd>
<dt>dae_threshold</dt>
<dd><p>An allelic expression (AE) threshold (in log-scale). A
sample showing an absolute AE greater than <code>dae_threshold</code> is considered
"technically" <em>differential allelic expressed</em>, meaning that the
imbalance observed is not below the limit of detection. Adjustment made to
this parameter should depend on the experimental assay sensitivity.</p></dd>
<dt>min_n_hom</dt>
<dd><p>Minimum number of samples in the homozygous group to be
considered eligible for statistical testing.</p></dd>
<dt>min_n_het</dt>
<dd><p>Minimum number of samples in the heterozygous group to be
considered eligible for statistical testing.</p></dd>
</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A data frame of two columns:</p><dl><dt>pvalue</dt>
<dd><p>The p-value associated with the statistical test, if performed; otherwise, <code>NA</code>.</p></dd>

<dt>case</dt>
<dd><p>One of the four possible cases depicted in the figure above.
The identified case depends on the number of samples for each group and on
the pattern of the allelic expression (AE) ratios of the DAE SNP:</p><dl><dt><code>1</code></dt>
<dd><p><strong>Case 1</strong>: the number of samples for which the candidate SNP is
heterozygous and DAE SNP expression is available is below the minimum
eligibility criterion <code>min_n_het</code>.</p></dd>

<dt><code>2</code></dt>
<dd><p><strong>Case 2</strong>: (i) the number of samples for which the candidate SNP is
homozygous is below <code>min_n_hom</code>, and also, (ii) the values of the DAE
SNP AE ratios are not all either below or above the <code>dae_threshold</code>.</p></dd>

<dt><code>3</code></dt>
<dd><p><strong>Case 3</strong>: the number of samples for which the candidate SNP is
homozygous is below <code>min_n_hom</code>, however, the values of the DAE
SNP AE ratios are all either below or above the <code>dae_threshold</code>. This
case is followed by a one-sample Wilcox test whose null hypothesis is that
the AE ratios are zero.</p></dd>

<dt><code>4</code></dt>
<dd><p><strong>Case 4</strong>: when both a minimum number of heterozygous and
homozygous samples for the candidate SNP are available, then a Wilcox test
is applied that compares the two groups. The AE ratios are first
transformed to absolute values. This is because we want to test departure
from zero in either direction. The null hypothesis is that absolute AE
ratios for the heterozygous group is less than or equal to the homozygous
group.</p></dd>


</dl></dd>


</dl></div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Case 1</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># The number of heterozygous samples (n = 1) does not meet the minimum</span></span>
<span class="r-in"><span class="co"># required (default is `min_n_het` = 2). Hence, no statistical test is</span></span>
<span class="r-in"><span class="co"># applied, the `pvalue` is `NA` and `case` is `1`.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">1.3</span> , <span class="fl">1.5</span>, <span class="fl">0.4</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fl">0.3</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     NA    1</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-1.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Be stricter with the number of heterozygous samples, e.g. `min_n_het = 4L`.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">2.1</span>, <span class="fl">3.2</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span>, min_n_het <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     NA    1</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-2.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Case 2</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># The number of homozygous samples does not meet the minimum requirement</span></span>
<span class="r-in"><span class="co"># (default is `min_n_hom` = 2) and also the AE ratios are not either all</span></span>
<span class="r-in"><span class="co"># above or all below the DAE threshold (`dae_threshold`).</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.1</span>, <span class="fl">1.3</span> , <span class="fl">1.5</span>, <span class="fl">0.4</span>, <span class="fl">0.59</span>, <span class="fl">0.67</span>, <span class="fl">0.89</span>, <span class="fl">1.35</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1     NA    2</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-3.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Case 3</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># The number of homozygous samples does not meet the minimum requirement</span></span>
<span class="r-in"><span class="co"># (default is `min_n_hom` = 2) but the AE ratios are either all above or all</span></span>
<span class="r-in"><span class="co"># below the DAE threshold (default is `dae_threshold = log2(1.5)`). Hence, a</span></span>
<span class="r-in"><span class="co"># one-sample Wilcox test is applied to test the null hypothesis that the</span></span>
<span class="r-in"><span class="co"># AE ratios for the heterozygous group is significantly different from zero.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2.8</span>, <span class="fl">3.1</span>, <span class="fl">1.3</span> , <span class="fl">1.5</span>, <span class="fl">2.3</span>, <span class="fl">0.59</span>, <span class="fl">0.67</span>, <span class="fl">0.89</span>, <span class="fl">1.35</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.009151689    3</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-4.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Case 4</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Both the number of homozygous and heterozygous samples are equal or</span></span>
<span class="r-in"><span class="co"># above the minima defined by `min_n_hom` and `min_n_het`. Hence, a</span></span>
<span class="r-in"><span class="co"># two-sample Wilcox test is applied to test the null hypothesis that the</span></span>
<span class="r-in"><span class="co"># absolute AE ratios for the heterozygous group is less than or equal to</span></span>
<span class="r-in"><span class="co"># those of the homozygous group.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="co"># Example: Both groups show imbalance above zero, i.e. show preference for</span></span>
<span class="r-in"><span class="co"># the same allele. However, the imbalance magnitude of the heterozygous group</span></span>
<span class="r-in"><span class="co"># is clearly below that of the homozygous group, resulting in a</span></span>
<span class="r-in"><span class="co"># non-significant p-value.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.9</span>, <span class="fl">2.1</span>, <span class="fl">2</span> , <span class="fl">1.5</span>, <span class="fl">1.4</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.6</span>, <span class="fl">0.7</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.9914637    4</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-5.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Example: the exact reverse of the previous example.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.6</span>, <span class="fl">0.7</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.9</span>, <span class="fl">2.1</span>, <span class="fl">2</span> , <span class="fl">1.5</span>, <span class="fl">1.4</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.01844421    4</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-6.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Example: the heterozygous group clearly shows greater imbalance (greater</span></span>
<span class="r-in"><span class="co"># departure from zero) than the homozygous group, resulting in a significant</span></span>
<span class="r-in"><span class="co"># p-value.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.3</span>, <span class="fl">0.2</span> , <span class="fl">0.21</span>, <span class="fl">0.15</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.8</span>, <span class="fl">1.2</span>, <span class="op">-</span><span class="fl">1.5</span>, <span class="op">-</span><span class="fl">3</span>, <span class="op">-</span><span class="fl">2.5</span>, <span class="op">-</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2.7</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.001676718    4</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-7.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
<span class="r-in"><span class="co"># Example: the exact reverse of the previous example.</span></span>
<span class="r-in"><span class="co">#</span></span>
<span class="r-in"><span class="va">ae_hom</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.8</span>, <span class="fl">1.2</span>, <span class="op">-</span><span class="fl">1.5</span>, <span class="op">-</span><span class="fl">3</span>, <span class="op">-</span><span class="fl">2.5</span>, <span class="op">-</span><span class="fl">1</span>, <span class="fl">2</span>, <span class="fl">2.7</span><span class="op">)</span></span>
<span class="r-in"><span class="va">ae_het</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.1</span>, <span class="fl">0.3</span>, <span class="fl">0.2</span> , <span class="fl">0.21</span>, <span class="fl">0.15</span><span class="op">)</span></span>
<span class="r-in"><span class="fu">daeqtl_test</span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      pvalue case</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.9989177    4</span>
<span class="r-in"><span class="fu"><a href="daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>ae_hom <span class="op">=</span> <span class="va">ae_hom</span>, ae_het <span class="op">=</span> <span class="va">ae_het</span><span class="op">)</span></span>
<span class="r-plt img"><img src="daeqtl_test-8.png" alt="" width="700" height="433"></span>
<span class="r-in"></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, Joana M Xavier, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://maialab.org/daeqtlr/reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer></div>

  

  

  </body></html>

