<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="daeqtlr">
<title>Introduction to daeqtlr • daeqtlr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to daeqtlr">
<meta property="og:description" content="daeqtlr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">daeqtlr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/daeqtlr.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="daeqtlr_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to daeqtlr</h1>
            
      
      
      <div class="d-none name"><code>daeqtlr.Rmd</code></div>
    </div>

    
    
<p>The goal of <a href="http://maialab.org/daeqtlr" class="external-link">daeqtlr</a> is to provide a minimum set of routines to perform DAEQTL mapping.</p>
<p>In this vignette we’ll show you how to perform DAEQTL mapping using an example data set. But before that let’s recap some key concepts underlying this analysis.</p>
<div class="section level2">
<h2 id="concepts">Concepts<a class="anchor" aria-label="anchor" href="#concepts"></a>
</h2>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p><a href="http://maialab.org/daeqtlr" class="external-link">daeqtlr</a> is bundled with an example data set that consists of three files:</p>
<ul>
<li>
<code>snp_pairs.csv</code>: A table of SNP pairs to be tested for association.</li>
<li>
<code>zygosity.csv</code>: A table providing the zygosity levels (homozygous or heterozygous) for each SNP/biological sample combination.</li>
<li>
<code>ae.csv</code>: Allelic expression (AE) ratios for each DAE SNP / biological sample combination.</li>
</ul>
<p>To easily read the bundled files you can use the function <code><a href="../reference/daeqtlr_example.html">daeqtlr_example()</a></code> to retrieve the path to each file, e.g. the path to <code>snp_pairs.csv</code> is:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"snp_pairs.csv"</span><span class="op">)</span>
<span class="co">#&gt; [1] "/home/runner/work/_temp/Library/daeqtlr/extdata/snp_pairs.csv"</span></code></pre></div>
<p>To import the data into R we provide a set of <code>read_*</code> functions:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snp_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_snp_pairs.html">read_snp_pairs</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"snp_pairs.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="va">zygosity</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_snp_zygosity.html">read_snp_zygosity</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"zygosity.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="va">ae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_ae_ratios.html">read_ae_ratios</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"ae.csv"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="snp-pairs">SNP pairs<a class="anchor" aria-label="anchor" href="#snp-pairs"></a>
</h3>
<p>The SNP pairs table indicates the pairs of SNPs (DAE SNP and candidate SNP) that will be tested for statistical association. Each row is for a pair. If this is not your starting point and you need to assemble this set of pairs first from a list of SNP, namely by looking for neighboring SNPs by genomic window, please read <code>vignette("snp-pairs")</code>.</p>
<p>The function <code><a href="../reference/read_snp_pairs.html">read_snp_pairs()</a></code> expects a path to a CSV file containing five columns:</p>
<ol style="list-style-type: decimal">
<li>
<code>dae_snp</code>: The id of the DAE SNP.</li>
<li>
<code>candidate_snp</code>: The id of the candidate DAEQTL SNP, or simply candidate SNP.</li>
<li>
<code>chromosome</code>: The chromosome name.</li>
<li>
<code>dae_snp_position</code>: The genomic position of the DAE SNP.</li>
<li>
<code>candidate_snp_position</code>: The genomic position of the candidate SNP.</li>
</ol>
<p>These columns are expected in this order. The actual column names in the header of the file are ignored and imported into R as indicated above. <code><a href="../reference/read_snp_pairs.html">read_snp_pairs()</a></code> will read the file with <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">data.table::fread()</a></code> and return a data table object.</p>
<p>In this example data set, all SNPs have dummy identifiers as should be clear from the non-valid rs identifiers: note the inclusion of a character <code>"X"</code> between the <code>"rs"</code> prefix and the SNP number.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First 10 pairs</span>
<span class="va">snp_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;     dae_snp candidate_snp chromosome dae_snp_position candidate_snp_position</span>
<span class="co">#&gt;  1:  rsX099        rsX008          1        180709100              180682638</span>
<span class="co">#&gt;  2:  rsX099        rsX011          1        180709100              181006806</span>
<span class="co">#&gt;  3:  rsX099        rsX012          1        180709100              181005024</span>
<span class="co">#&gt;  4:  rsX099        rsX013          1        180709100              180709233</span>
<span class="co">#&gt;  5:  rsX099        rsX015          1        180709100              181114464</span>
<span class="co">#&gt;  6:  rsX099        rsX020          1        180709100              181020238</span>
<span class="co">#&gt;  7:  rsX099        rsX021          1        180709100              181060085</span>
<span class="co">#&gt;  8:  rsX099        rsX030          1        180709100              180744030</span>
<span class="co">#&gt;  9:  rsX099        rsX031          1        180709100              181058720</span>
<span class="co">#&gt; 10:  rsX099        rsX037          1        180709100              181106008</span>

<span class="co"># Total number of pairs</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">snp_pairs</span><span class="op">)</span>
<span class="co">#&gt; [1] 200</span>

<span class="co"># Chromosomes</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">snp_pairs</span><span class="op">$</span><span class="va">chromosome</span><span class="op">)</span>
<span class="co">#&gt; [1] "1"  "19" "4"</span></code></pre></div>
<p>There are 200 pairs, scattered across 3 chromosomes: 1, 19, and 4. This means that the DAEQTL mapping will consist (potentially) of 200 statistical tests.</p>
</div>
<div class="section level3">
<h3 id="zygosity">Zygosity<a class="anchor" aria-label="anchor" href="#zygosity"></a>
</h3>
<p>The <code>zygosity</code> data table consists of the zygosity levels for the SNP / sample combination. Samples are also dummy and are therefore generically named <code>"s01"</code>, <code>"s02"</code>, etc. The values <code>"hom"</code> and <code>"het"</code> stand for homozygous and heterozygous, respectively.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First 10 SNPs, first 4 samples (first column is the SNP identifier)</span>
<span class="va">zygosity</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;        snp s01 s02 s03 s04</span>
<span class="co">#&gt;  1: rsX001 hom hom hom het</span>
<span class="co">#&gt;  2: rsX002 hom hom hom hom</span>
<span class="co">#&gt;  3: rsX003 het hom hom het</span>
<span class="co">#&gt;  4: rsX004 hom hom hom het</span>
<span class="co">#&gt;  5: rsX005 het hom het het</span>
<span class="co">#&gt;  6: rsX006 hom het het hom</span>
<span class="co">#&gt;  7: rsX007 hom hom hom hom</span>
<span class="co">#&gt;  8: rsX008 hom hom hom hom</span>
<span class="co">#&gt;  9: rsX009 het het hom hom</span>
<span class="co">#&gt; 10: rsX010 het hom het het</span>

<span class="co"># Number of genotyped (determined zygosity) SNPs</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">zygosity</span><span class="op">)</span>
<span class="co">#&gt; [1] 66</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="allelic-expression-ratios">Allelic expression ratios<a class="anchor" aria-label="anchor" href="#allelic-expression-ratios"></a>
</h3>
<p>The <code>ae</code> data table contains the log2 of the ratio of one of the alleles over the other. This metric is also known as the M-value <span class="citation">1</span>. An alternative metric is the Beta-value, i.e. the relative expression of one allele. In this example data set, the <code>ae</code> data table comprises M-values because it has been shown to have more desirable properties for statistical testing <span class="citation">1</span>. However, if you find Beta-values more intuitive and hence preferable for reporting results you may use the function <code><a href="../reference/m2b.html">m2b()</a></code> for conversion (and <code><a href="../reference/b2m.html">b2m()</a></code> for the reverse operation).</p>
<p>M-values can vary between <code>-Inf</code> and <code>Inf</code>, where zero means balanced allelic expression.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First 4 samples (first column is the SNP identifier)</span>
<span class="va">ae</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;     dae_snp         s01         s02         s03         s04</span>
<span class="co">#&gt;  1:  rsX031  0.43276128  1.28668700  0.60065229  0.36049254</span>
<span class="co">#&gt;  2:  rsX058          NA  1.89762256  1.38378226 -0.92397713</span>
<span class="co">#&gt;  3:  rsX049  0.46547396  0.64117418  0.49471933 -0.24431776</span>
<span class="co">#&gt;  4:  rsX051  0.23175630  0.51505900          NA          NA</span>
<span class="co">#&gt;  5:  rsX038 -0.47084220  0.34195080  0.50772590 -0.08951668</span>
<span class="co">#&gt;  6:  rsX041  0.16917890 -0.85413270  0.35275210  2.55589100</span>
<span class="co">#&gt;  7:  rsX020  2.46806900          NA  2.81560000          NA</span>
<span class="co">#&gt;  8:  rsX019  0.34176550  0.02681199  0.09909749 -0.27985290</span>
<span class="co">#&gt;  9:  rsX060  0.07940053 -0.61600390 -0.73958170  1.01771300</span>
<span class="co">#&gt; 10:  rsX048  1.31660300  0.84591350  1.24441500  1.31580000</span>
<span class="co">#&gt; 11:  rsX059  0.11945310  0.65581140          NA  0.50474010</span>

<span class="co"># Number of SNPs with measured allelic expression</span>
<span class="co"># These are the DAE SNPs</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ae</span><span class="op">)</span>
<span class="co">#&gt; [1] 11</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="daeqtl-mapping">DAEQTL mapping<a class="anchor" aria-label="anchor" href="#daeqtl-mapping"></a>
</h2>
<p>To perform DAEQTL mapping you use the function <code><a href="../reference/daeqtl_mapping.html">daeqtl_mapping()</a></code>. The P-value associated with the statistical association will be appended as a new column to the data table <code>snp_pairs</code> by reference, i.e. the <code>snp_pairs</code> is changed in-place, no extra copy is created. Here we associate a new variable <code>mapping_results</code> with the output of <code><a href="../reference/daeqtl_mapping.html">daeqtl_mapping()</a></code> but it refers to the same data in memory as the new updated <code>snp_pairs</code> data table.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># `snp_pairs` is updated by reference, so `snp_pairs` and `mapping_results`</span>
<span class="co"># refer to the same object in memory.</span>
<span class="va">mapping_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/daeqtl_mapping.html">daeqtl_mapping</a></span><span class="op">(</span>snp_pairs <span class="op">=</span> <span class="va">snp_pairs</span>, zygosity <span class="op">=</span> <span class="va">zygosity</span>, ae <span class="op">=</span> <span class="va">ae</span><span class="op">)</span>

<span class="co"># Omiting here the columns `dae_snp_position` and `candidate_snp_position`</span>
<span class="co"># and showing only the first 30 pairs for brevity.</span>
<span class="va">mapping_results</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'dae_snp_position'</span>, <span class="st">'candidate_snp_position'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt;     dae_snp candidate_snp chromosome    pvalue case</span>
<span class="co">#&gt;  1:  rsX019        rsX002         19 0.8331668    4</span>
<span class="co">#&gt;  2:  rsX019        rsX003         19 0.1348031    4</span>
<span class="co">#&gt;  3:  rsX019        rsX005         19 0.9280741    4</span>
<span class="co">#&gt;  4:  rsX019        rsX010         19 0.9280741    4</span>
<span class="co">#&gt;  5:  rsX019        rsX014         19 0.1348031    4</span>
<span class="co">#&gt;  6:  rsX019        rsX016         19        NA    2</span>
<span class="co">#&gt;  7:  rsX019        rsX017         19        NA    2</span>
<span class="co">#&gt;  8:  rsX019        rsX018         19 0.9255817    4</span>
<span class="co">#&gt;  9:  rsX019        rsX022         19 0.3994533    4</span>
<span class="co">#&gt; 10:  rsX019        rsX023         19 0.3736310    4</span>
<span class="co">#&gt; 11:  rsX019        rsX024         19 0.8813157    4</span>
<span class="co">#&gt; 12:  rsX019        rsX025         19        NA    1</span>
<span class="co">#&gt; 13:  rsX019        rsX026         19        NA    1</span>
<span class="co">#&gt; 14:  rsX019        rsX043         19        NA    1</span>
<span class="co">#&gt; 15:  rsX019        rsX045         19        NA    2</span>
<span class="co">#&gt; 16:  rsX019        rsX046         19        NA    1</span>
<span class="co">#&gt; 17:  rsX019        rsX047         19        NA    2</span>
<span class="co">#&gt; 18:  rsX019        rsX048         19        NA    2</span>
<span class="co">#&gt; 19:  rsX019        rsX050         19        NA    2</span>
<span class="co">#&gt; 20:  rsX019        rsX051         19        NA    2</span>
<span class="co">#&gt; 21:  rsX019        rsX053         19        NA    2</span>
<span class="co">#&gt; 22:  rsX019        rsX054         19        NA    2</span>
<span class="co">#&gt; 23:  rsX019        rsX060         19 0.1348031    4</span>
<span class="co">#&gt; 24:  rsX019        rsX064         19        NA    1</span>
<span class="co">#&gt; 25:  rsX020        rsX008          1 0.1226391    4</span>
<span class="co">#&gt; 26:  rsX020        rsX011          1 0.1003482    3</span>
<span class="co">#&gt; 27:  rsX020        rsX012          1 0.1003482    3</span>
<span class="co">#&gt; 28:  rsX020        rsX013          1        NA    1</span>
<span class="co">#&gt; 29:  rsX020        rsX015          1 0.3492677    4</span>
<span class="co">#&gt; 30:  rsX020        rsX021          1 0.1814492    3</span>
<span class="co">#&gt;     dae_snp candidate_snp chromosome    pvalue case</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<!-- References will be automatically included here -->
<div id="refs" class="references">
<div id="ref-Du.BB.2010">
<p>1. Du, P. <em>et al.</em> Comparison of Beta-value and M-value methods for quantifying methylation levels by microarray analysis. <em>BMC Bioinformatics</em> <strong>11</strong>, (2010).</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, Joana M Xavier, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://maialab.org/daeqtlr/reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
