<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="daeqtlr">
<title>Introduction to daeqtlr • daeqtlr</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to daeqtlr">
<meta property="og:description" content="daeqtlr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">daeqtlr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/daeqtlr.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="daeqtlr_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to daeqtlr</h1>
            
      
      
      <div class="d-none name"><code>daeqtlr.Rmd</code></div>
    </div>

    
    
<p>The goal of <a href="http://maialab.org/daeqtlr" class="external-link">daeqtlr</a> is to provide a minimum set of routines to perform DAEQTL mapping.</p>
<p>In this vignette we’ll show you how to perform DAEQTL mapping using an example data set. But before that let’s recap some key concepts underlying this analysis.</p>
<div class="section level2">
<h2 id="concepts">Concepts<a class="anchor" aria-label="anchor" href="#concepts"></a>
</h2>
<ul>
<li><p>DAE SNP: SNP on which the allelic expression (AE) ratio is measured. AE is the quantitative trait used as one of the variables in the statistical association test when performing DAEQTL mapping. AE ratios are only sensible for DAE SNPs that are heterozygous.</p></li>
<li><p>Candidate SNP: SNP candidate for DAE quantitative trait loci (DAEQTL). These are the SNPs whose zygosity level is used as the other variable in the statistical association test.</p></li>
<li><p>Zygosity: degree to which both copies of a locus in two homologous chromosomes have the same genetic sequence or not. If both copies are the same, then the locus is homozygous, if they are different, the loci are heterozygous.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p><a href="http://maialab.org/daeqtlr" class="external-link">daeqtlr</a> is bundled with an example data set that consists of three files:</p>
<ul>
<li>
<code>snp_pairs.csv</code>: A table of SNP pairs to be tested for association.</li>
<li>
<code>zygosity.csv</code>: A table providing the zygosity levels (homozygous or heterozygous) for each SNP/biological sample combination.</li>
<li>
<code>ae.csv</code>: Allelic expression (AE) ratios for each DAE SNP / biological sample combination.</li>
</ul>
<p>To easily read the bundled files you can use the function <code><a href="../reference/daeqtlr_example.html">daeqtlr_example()</a></code> to retrieve the path to each file, e.g. the path to <code>snp_pairs.csv</code> is:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"snp_pairs.csv"</span><span class="op">)</span>
<span class="co">#&gt; [1] "/home/runner/work/_temp/Library/daeqtlr/extdata/snp_pairs.csv"</span></code></pre></div>
<p>To import the data into R we provide a set of <code>read_*</code> functions:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">snp_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_snp_pairs.html">read_snp_pairs</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"snp_pairs.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="va">zygosity</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_snp_zygosity.html">read_snp_zygosity</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"zygosity.csv"</span><span class="op">)</span><span class="op">)</span>
<span class="va">ae</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_ae_ratios.html">read_ae_ratios</a></span><span class="op">(</span>file <span class="op">=</span> <span class="fu"><a href="../reference/daeqtlr_example.html">daeqtlr_example</a></span><span class="op">(</span><span class="st">"ae.csv"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="section level3">
<h3 id="snp-pairs">SNP pairs<a class="anchor" aria-label="anchor" href="#snp-pairs"></a>
</h3>
<p>The SNP pairs table indicates the pairs of SNPs (DAE SNP and candidate SNP) that will be tested for statistical association. Each row is for a pair. If this is not your starting point and you need to assemble this set of pairs first from a list of SNP, namely by looking for neighboring SNPs by genomic window, please read <code>vignette("snp-pairs")</code>.</p>
<p>The function <code><a href="../reference/read_snp_pairs.html">read_snp_pairs()</a></code> expects a path to a CSV file containing five columns:</p>
<ol style="list-style-type: decimal">
<li>
<code>dae_snp</code>: The id of the DAE SNP.</li>
<li>
<code>candidate_snp</code>: The id of the candidate DAEQTL SNP, or simply candidate SNP.</li>
<li>
<code>chromosome</code>: The chromosome name.</li>
<li>
<code>dae_snp_position</code>: The genomic position of the DAE SNP.</li>
<li>
<code>candidate_snp_position</code>: The genomic position of the candidate SNP.</li>
</ol>
<p>These columns are expected in this order. The actual column names in the header of the file are ignored and imported into R as indicated above. <code><a href="../reference/read_snp_pairs.html">read_snp_pairs()</a></code> will read the file with <code><a href="https://Rdatatable.gitlab.io/data.table/reference/fread.html" class="external-link">data.table::fread()</a></code> and return a data table object.</p>
<p>In this example data set, all SNPs have dummy identifiers as should be clear from the non-valid rs identifiers: note the inclusion of a character <code>"X"</code> between the <code>"rs"</code> prefix and the SNP number.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First 10 pairs</span>
<span class="va">snp_pairs</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span>
<span class="co">#&gt;     dae_snp candidate_snp chromosome dae_snp_position candidate_snp_position</span>
<span class="co">#&gt;  1:  rsX099        rsX008          1        180709100              180682638</span>
<span class="co">#&gt;  2:  rsX099        rsX011          1        180709100              181006806</span>
<span class="co">#&gt;  3:  rsX099        rsX012          1        180709100              181005024</span>
<span class="co">#&gt;  4:  rsX099        rsX013          1        180709100              180709233</span>
<span class="co">#&gt;  5:  rsX099        rsX015          1        180709100              181114464</span>
<span class="co">#&gt;  6:  rsX099        rsX020          1        180709100              181020238</span>
<span class="co">#&gt;  7:  rsX099        rsX021          1        180709100              181060085</span>
<span class="co">#&gt;  8:  rsX099        rsX030          1        180709100              180744030</span>
<span class="co">#&gt;  9:  rsX099        rsX031          1        180709100              181058720</span>
<span class="co">#&gt; 10:  rsX099        rsX037          1        180709100              181106008</span>

<span class="co"># Total number of pairs</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">snp_pairs</span><span class="op">)</span>
<span class="co">#&gt; [1] 200</span>

<span class="co"># Chromosomes</span>
<span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">snp_pairs</span><span class="op">$</span><span class="va">chromosome</span><span class="op">)</span>
<span class="co">#&gt; [1] "1"  "19" "4"</span></code></pre></div>
<p>There are 200 pairs, scattered across 3 chromosomes: 1, 19, and 4. This means that the DAEQTL mapping will consist (potentially) of 200 statistical tests.</p>
</div>
<div class="section level3">
<h3 id="zygosity">Zygosity<a class="anchor" aria-label="anchor" href="#zygosity"></a>
</h3>
<p>The <code>zygosity</code> data table consists of the zygosity levels for the SNP / sample combination. Samples are also dummy and are therefore generically named <code>"s01"</code>, <code>"s02"</code>, etc. The values <code>"hom"</code> and <code>"het"</code> stand for homozygous and heterozygous, respectively.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># First 10 SNPs, first 15 samples (first column is the SNP identifier)</span>
<span class="va">zygosity</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">16</span><span class="op">]</span>
<span class="co">#&gt;        snp s01 s02 s03 s04 s05 s06 s07 s08 s09 s10 s11 s12 s13 s14 s15</span>
<span class="co">#&gt;  1: rsX001 hom hom hom het hom hom hom hom hom hom hom hom het hom hom</span>
<span class="co">#&gt;  2: rsX002 hom hom hom hom hom hom hom hom hom hom het hom hom hom hom</span>
<span class="co">#&gt;  3: rsX003 het hom hom het het hom hom hom hom hom hom het hom hom hom</span>
<span class="co">#&gt;  4: rsX004 hom hom hom het het hom hom het hom hom het het hom hom hom</span>
<span class="co">#&gt;  5: rsX005 het hom het het hom het het het hom hom hom hom het hom het</span>
<span class="co">#&gt;  6: rsX006 hom het het hom hom het hom het hom het het hom het hom het</span>
<span class="co">#&gt;  7: rsX007 hom hom hom hom hom hom hom hom hom hom hom hom hom hom hom</span>
<span class="co">#&gt;  8: rsX008 hom hom hom hom hom hom het hom het het hom het het het het</span>
<span class="co">#&gt;  9: rsX009 het het hom hom hom het hom hom hom hom hom hom hom hom hom</span>
<span class="co">#&gt; 10: rsX010 het hom het het hom het het het hom hom hom hom het hom het</span>

<span class="co"># Number of genotyped (determined zygosity) SNPs</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">zygosity</span><span class="op">)</span>
<span class="co">#&gt; [1] 66</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="allelic-expression-ratios-m-values">Allelic expression ratios (M-values)<a class="anchor" aria-label="anchor" href="#allelic-expression-ratios-m-values"></a>
</h3>
<p>The <code>ae</code> data table contains the <span class="math inline">\(\log_2\)</span> of the ratio of the expression of one of the alleles over that of the other. This metric is also known as the M-value<span class="citation"><sup>1</sup></span>. In the <a href="http://maialab.org/daeqtlr" class="external-link">daeqtlr</a> documentation we typically refer to M-values by <em>log-ratios</em> or <em>AE ratios</em>.</p>
<p><span class="math display">\[\text{M} = \log_2 \frac{x_{A}}{x_{B}}\]</span></p>
<p>where <span class="math inline">\(x_{A}\)</span> and <span class="math inline">\(x_{B}\)</span> are the allelic expression of the A and B alleles, respectively. These can be microarray intensities or sequencing counts or any other measure that bears a linear relationship with allelic expression.</p>
<p>An alternative metric is the Beta-value <span class="math inline">\((\beta)\)</span>, i.e. the relative expression of one allele in the total of the two alleles’ expression:</p>
<p><span class="math display">\[\beta = \frac{x_{A}}{x_{A} + x_{B}}\]</span></p>
<p>In this example data set, the <code>ae</code> data table comprises M-values because it has been shown to have more desirable properties for statistical testing<span class="citation"><sup>1</sup></span>. However, if you find Beta-values more intuitive and hence preferable for reporting results you may use the function <code><a href="../reference/m2b.html">m2b()</a></code> for conversion (and <code><a href="../reference/m2b.html">b2m()</a></code> for the reverse operation).</p>
<p>M-values can vary between <code>-Inf</code>, if only one of the alleles is expressed, and <code>Inf</code>, if only the other allele is expressed. An M-value of zero means balanced allelic expression (equal expression of both alleles).</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Number of DAE SNPs, i.e. SNPs with measured allelic expression</span>
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">ae</span><span class="op">)</span>
<span class="co">#&gt; [1] 11</span>

<span class="co"># First 5 samples (first column is the SNP identifier)</span>
<span class="va">ae</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>
<span class="co">#&gt;     dae_snp         s01         s02         s03         s04         s05</span>
<span class="co">#&gt;  1:  rsX019  0.34176550  0.02681199  0.09909749 -0.27985290 -0.13830710</span>
<span class="co">#&gt;  2:  rsX020  2.46806900          NA  2.81560000          NA  1.17734500</span>
<span class="co">#&gt;  3:  rsX031  0.43276128  1.28668700  0.60065229  0.36049254 -1.31717287</span>
<span class="co">#&gt;  4:  rsX038 -0.47084220  0.34195080  0.50772590 -0.08951668  0.07892864</span>
<span class="co">#&gt;  5:  rsX041  0.16917890 -0.85413270  0.35275210  2.55589100 -1.00207400</span>
<span class="co">#&gt;  6:  rsX048  1.31660300  0.84591350  1.24441500  1.31580000  0.53643270</span>
<span class="co">#&gt;  7:  rsX049  0.46547396  0.64117418  0.49471933 -0.24431776  0.48254007</span>
<span class="co">#&gt;  8:  rsX051  0.23175630  0.51505900          NA          NA  0.80177990</span>
<span class="co">#&gt;  9:  rsX058          NA  1.89762256  1.38378226 -0.92397713  2.41020835</span>
<span class="co">#&gt; 10:  rsX059  0.11945310  0.65581140          NA  0.50474010  0.48633720</span>
<span class="co">#&gt; 11:  rsX060  0.07940053 -0.61600390 -0.73958170  1.01771300 -0.31974190</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="daeqtl-mapping">DAEQTL mapping<a class="anchor" aria-label="anchor" href="#daeqtl-mapping"></a>
</h2>
<p>To perform DAEQTL mapping you use the function <code><a href="../reference/daeqtl_mapping.html">daeqtl_mapping()</a></code>. The P-value associated with the statistical association will be appended as a new column to the data table <code>snp_pairs</code> by reference, i.e. the <code>snp_pairs</code> is changed in-place, no extra copy is created. Here we associate a new variable <code>mapping_results</code> with the output of <code><a href="../reference/daeqtl_mapping.html">daeqtl_mapping()</a></code> but it refers to the same data in memory as the new updated <code>snp_pairs</code> data table.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># `snp_pairs` is updated by reference, so `snp_pairs` and `mapping_results`</span>
<span class="co"># refer to the same object in memory.</span>
<span class="va">mapping_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/daeqtl_mapping.html">daeqtl_mapping</a></span><span class="op">(</span>snp_pairs <span class="op">=</span> <span class="va">snp_pairs</span>,
                                  zygosity <span class="op">=</span> <span class="va">zygosity</span>,
                                  ae <span class="op">=</span> <span class="va">ae</span><span class="op">)</span>

<span class="co"># Omiting here the columns `dae_snp_position` and `candidate_snp_position`</span>
<span class="co"># and showing only the first 5 pairs for brevity.</span>
<span class="va">mapping_results</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'dae_snp_position'</span>, <span class="st">'candidate_snp_position'</span><span class="op">)</span><span class="op">]</span>
<span class="co">#&gt;    dae_snp candidate_snp chromosome    pvalue case</span>
<span class="co">#&gt; 1:  rsX019        rsX002         19 0.8331668    4</span>
<span class="co">#&gt; 2:  rsX019        rsX003         19 0.1348031    4</span>
<span class="co">#&gt; 3:  rsX019        rsX005         19 0.9280741    4</span>
<span class="co">#&gt; 4:  rsX019        rsX010         19 0.9280741    4</span>
<span class="co">#&gt; 5:  rsX019        rsX014         19 0.1348031    4</span></code></pre></div>
<p>Now let us sort by ascending P value to check the pairs that show the most strong evidence of association.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mapping_results</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'dae_snp_position'</span>, <span class="st">'candidate_snp_position'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">pvalue</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span>
<span class="co">#&gt;     dae_snp candidate_snp chromosome      pvalue case</span>
<span class="co">#&gt;  1:  rsX059        rsX057          4 0.003857293    3</span>
<span class="co">#&gt;  2:  rsX031        rsX062          1 0.004554027    4</span>
<span class="co">#&gt;  3:  rsX031        rsX039          1 0.011581429    4</span>
<span class="co">#&gt;  4:  rsX038        rsX035          4 0.019107190    4</span>
<span class="co">#&gt;  5:  rsX041        rsX032          4 0.033376508    4</span>
<span class="co">#&gt;  6:  rsX058        rsX057          4 0.038642034    4</span>
<span class="co">#&gt;  7:  rsX058        rsX059          4 0.038642034    4</span>
<span class="co">#&gt;  8:  rsX041        rsX009          4 0.041884336    4</span>
<span class="co">#&gt;  9:  rsX051        rsX023         19 0.046695338    4</span>
<span class="co">#&gt; 10:  rsX060        rsX003         19 0.059058229    3</span></code></pre></div>
<div class="section level3">
<h3 id="rsx059-and-rsx057">rsX059 and rsX057<a class="anchor" aria-label="anchor" href="#rsx059-and-rsx057"></a>
</h3>
<p>TODO: Explanation of case 3.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DAEQTL plot for the pair rsX059/rsX057</span>
<span class="fu"><a href="../reference/daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>
  ae_hom <span class="op">=</span> <span class="fu"><a href="../reference/ae.html">ae_hom</a></span><span class="op">(</span><span class="st">'rsX059'</span>, <span class="st">'rsX057'</span>, <span class="va">zygosity</span>, <span class="va">ae</span><span class="op">)</span>,
  ae_het <span class="op">=</span> <span class="fu"><a href="../reference/ae.html">ae_het</a></span><span class="op">(</span><span class="st">'rsX059'</span>, <span class="st">'rsX057'</span>, <span class="va">zygosity</span>, <span class="va">ae</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="daeqtlr_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div class="section level3">
<h3 id="rsx031-and-rsx062">rsX031 and rsX062<a class="anchor" aria-label="anchor" href="#rsx031-and-rsx062"></a>
</h3>
<p>TODO: Explanation of case 4.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># DAEQTL plot for the pair rsX031/rsX062</span>
<span class="fu"><a href="../reference/daeqtl_plot.html">daeqtl_plot</a></span><span class="op">(</span>
  ae_hom <span class="op">=</span> <span class="fu"><a href="../reference/ae.html">ae_hom</a></span><span class="op">(</span><span class="st">'rsX031'</span>, <span class="st">'rsX062'</span>, <span class="va">zygosity</span>, <span class="va">ae</span><span class="op">)</span>,
  ae_het <span class="op">=</span> <span class="fu"><a href="../reference/ae.html">ae_het</a></span><span class="op">(</span><span class="st">'rsX031'</span>, <span class="st">'rsX062'</span>, <span class="va">zygosity</span>, <span class="va">ae</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<p><img src="daeqtlr_files/figure-html/unnamed-chunk-10-1.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<!-- References will be automatically included here -->
<div id="refs" class="references">
<div id="ref-Du.BB.2010">
<p>1. Du, P. <em>et al.</em> Comparison of Beta-value and M-value methods for quantifying methylation levels by microarray analysis. <em>BMC Bioinformatics</em> <strong>11</strong>, (2010).</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://rmagno.eu/" class="external-link">Ramiro Magno</a>, Joana M Xavier, Ana-Teresa Maia, <a href="https://cintesis.eu/" class="external-link"><img src="https://maialab.org/daeqtlr/reference/figures/cintesis.svg" height="24"></a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
